packages/discovery-provider/src/tasks/entity_manager/entities/playlist.py

```diff
@@ -110,6 +111,57 @@ def update_playlist_routes_table(
     )
 
 
+def update_playlist_tracks_relations(
+    params: ManageEntityParameters, playlist_record: Playlist
+):
+    # Update the playlist_tracks_relations table
+    session = params.session
+    existing_playlist_tracks_relations = (
+        session.query(CollectionTrackRelation)
+        .filter(
+            CollectionTrackRelation.collection_id == params.entity_id,
+        )
+        .all()
+    )
+    existing_tracks = {
+        track.track_id: track for track in existing_playlist_tracks_relations
+    }
+    playlist = helpers.model_to_dictionary(playlist_record)
+    updated_track_ids = [
+        track["track"] for track in playlist["playlist_contents"]["track_ids"]
+    ]
+    params.logger.info(
+        f"playlists.py | Updating playlist tracks relations for {playlist['playlist_id']}"
+    )
+
+    # delete relations that previously existed but are not in the updated list
+    for relation in existing_playlist_tracks_relations:
+        if relation.track_id not in updated_track_ids:
+            relation.is_delete = True
+            relation.updated_at = params.block_datetime
+
+    # add row for each track that is not already in the table
```

Comment:
technically add or recover :)

Reply:
moved comment around, had a 2nd comment further down for recover case.